core/global/platform.gd

```diff
@@ -388,80 +392,179 @@ func _get_lscpu_info() -> Array:
 	return  output[0][0].split("\n") as Array
 
 
-# Reads system files and tools to fill out the GPUInfo
+## Reads system files and tools to fill out the GPUInfo
 func _read_gpu_info() -> GPUInfo:
+	var gpu_list: Array[GPUInfo]
 	var gpu_info := GPUInfo.new()
-	var gpu_raw := _get_glxinfo()
 
-	# Get the GPU Vendor, Model, and Driver
-	for param in gpu_raw:
-		var parts := param.split(" ", false) as Array
-		if parts.is_empty():
-			continue
-		if parts[0] == "OpenGL" and parts[1] == "vendor" and parts[2] == "string:":
-			parts.remove_at(2)
-			parts.remove_at(1)
-			parts.remove_at(0)
-			gpu_info.vendor = str(" ".join(parts))
-		if parts[0] == "OpenGL" and parts[1] == "renderer" and parts[2] == "string:":
-			parts.remove_at(2)
-			parts.remove_at(1)
-			parts.remove_at(0)
-			gpu_info.model = str(" ".join(parts))
-		if parts[0] == "OpenGL" and parts[1] == "version" and parts[2] == "string:":
-			parts.remove_at(2)
-			parts.remove_at(1)
-			parts.remove_at(0)
-			gpu_info.driver = str(" ".join(parts))
-	logger.debug("Found GPU: Vendor: " + gpu_info.vendor + "Model: " + gpu_info.model)
+	# Get the info reported by the current rendering driver.
+	# TODO: This is much more simple than glxinfo but we need
+	# to look into vulkaninfo as this can only gather the data
+	# for the currently active GPU. Vulkaninfo can provide data
+	# on all detected GPU devices.
+	match RenderingServer.get_video_adapter_vendor():
+		"AMD", "AuthenticAMD", 'AuthenticAMD Advanced Micro Devices, Inc.', "Advanced Micro Devices, Inc. [AMD/ATI]":
+			gpu_info.vendor = "AMD"
+		"Intel", "GenuineIntel", "Intel Corporation":
+			gpu_info.vendor = "Intel"
+		"Nvidia":
+			gpu_info.vendor = "Trash" # :D
+			logger.info("Nvidia devices are not suppored.")
+			return null
+		_:
+			logger.warn("Device vendor string not recognized: " + RenderingServer.get_video_adapter_vendor())
+			return null
+
+	gpu_info.model = RenderingServer.get_video_adapter_name()
+	gpu_info.driver = RenderingServer.get_video_adapter_api_version()
+
+	# TODO: This is bad and lazy. We need to identify a correct way to link the
+	# PCI Address to the rendering info
+	# Assume the first card is the one we are using.
+	cards = _get_gpu_cards()
+	if cards.size() <= 0:
+		logger.warn("GPU Data could not be derived.")
+		return null
+
+	logger.debug("Assuming " + cards[0].name + " is the correct card!")
+	gpu_info.card = cards[0]
```

Comment:
It looks like we can parse `glxinfo` to get the device ID of the currently used device:

![image](https://github.com/ShadowBlip/OpenGamepadUI/assets/376460/36826c84-4047-4311-9736-1babf53fe5c3)

`0x1636` is the device ID of my integrated graphics.

If we can pull this device ID out, we can match on `card.device` for the correct `Card` instance.

Reply:
Bet

Reply:
A function like this should work to parse that device id:

```gdscript

func get_current_card_device() -> String:
	var cmd := Command.new("glxinfo")
	if await cmd.execute() != OK:
		return ""
	
	for line in cmd.stdout.split("\n"):
		if not "Device: " in line:
			continue
		
		# Match on the last part of the string to get the device ID
		# E.g. Device: AMD Radeon Graphics (renoir, LLVM 16.0.6, DRM 3.54, 6.5.5-arch1-1) (0x1636)
		var regex := RegEx.new()
		regex.compile("0x(\\d+)\\)$")
		var result := regex.search(line)
		if not result:
			continue
		var match_str := result.get_string()

		return match_str.replace("0x", "").replace(")", "")

	return ""
```

Reply:
We should probably grab the vendor too to cleanly match on vendor + device:

```
    Vendor: AMD (0x1002)
    Device: AMD Radeon Graphics (renoir, LLVM 16.0.6, DRM 3.54, 6.5.5-arch1-1) (0x1636)
```